c7n/resources/ami.py

```diff
@@ -609,3 +609,72 @@ def process(self, resources, event=None):
                     continue
                 results.extend(f.result())
         return results
+
+
+@AMI.filter_registry.register('image-attribute')
+class ImageAttribute(ValueFilter):
+    """AMI Image Value Filter on a given image attribute.
+
+    Filters AMI's with the given AMI attribute
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: ami-unused-recently
+                resource: ami
+                filters:
+                  - type: image-attribute
+                    attribute: lastLaunchedTime
+                    key: "Value"
+                    op: gte
+                    value_type: age
+                    value: 30
+    """
+
+    valid_attrs = (
+        'kernel',
+        'ramdisk',
+        'launchPermissions',
+        'productCodes',
+        'blockDeviceMapping',
+        'sriovNetSupport',
+        'bootMode',
+        'tpmSupport',
+        'uefiData',
+        'lastLaunchedTime',
+        'imdsSupport'
+    )
+
+    schema = type_schema(
+        'image-attribute',
+        rinherit=ValueFilter.schema,
+        attribute={'enum': valid_attrs},
+        required=('attribute',))
+    schema_alias = False
+
+    def get_permissions(self):
+        return ('ec2:DescribeImageAttribute',)
+
+    def process(self, resources, event=None):
+        attribute = self.data['attribute']
+        self.get_image_attribute(resources, attribute)
+        return [resource for resource in resources
+                if self.match(resource['c7n:attribute-%s' % attribute])]
+
+    def get_image_attribute(self, resources, attribute):
+        client = local_session(
+            self.manager.session_factory).client('ec2')
+
+        for resource in resources:
+            image_id = resource['ImageId']
+            fetched_attribute = self.manager.retry(
+                client.describe_image_attribute,
+                ImageId=image_id,
+                Attribute=attribute)
+            keys = list(fetched_attribute.keys())
```

Comment:
Oh shoot you caught me slacking. 😅  This probably looks awfully similar to https://github.com/cloud-custodian/cloud-custodian/blob/master/c7n/resources/ec2.py#L2139-L2205.  I agree that it could be cleaner; will update.